atc/api/policychecker/checker.go

```diff
@@ -59,15 +59,15 @@ func (c *checker) Check(action string, acc accessor.Access, req *http.Request) (
 	case "application/json", "text/vnd.yaml", "text/yaml", "text/x-yaml", "application/x-yaml":
 		body, err := io.ReadAll(req.Body)
 		if err != nil {
-			return nil, err
+			return policy.PolicyCheckResult{}, err
```

Comment:
See my comments in the tests.
```suggestion
			return policy.PolicyCheckResult{
			    Status: policy.Block,
			    Reasons: fmt.Sprintf("I/O error: %s", err)
			}
```

Reply:
> See my comments in the tests.

sure, I will.. But my first reaction is no... if error is not nil one should never inspect the first object (the null value of the first return object)

Reply:
Any error from here should be handled by:
```
	result, err := h.policyChecker.Check(h.action, acc, r)
	if err != nil {
		w.WriteHeader(http.StatusBadRequest)
		fmt.Fprintf(w, "policy check error: %s", err.Error())
		return
	}
```